lib/matplotlib/image.py

```diff
@@ -1470,9 +1472,8 @@ def imread(fname, format=None):
     if isinstance(fname, str):
         parsed = urllib.parse.urlparse(fname)
         if len(parsed.scheme) > 1:  # Pillow doesn't handle URLs directly.
-            from urllib import request
             with urllib.request.urlopen(fname) as response:
-                return imread(response, format=ext)
+                return imread(BytesIO(response.read()), format=ext)
```

Comment:
Should we be defensive and only wrap if necessary?

```suggestion
                try:
                    return imread(response, format=ext)
                except (AttributeError, io.UnsupportedOperation):
                    # wrap HTTPResponse because does not support seek()
                    return imread(BytesIO(response.read()), format=ext)
```

Reply:
The first call might have half-read it though, and then we have no `seek` to go back...

Reply:
Good point.

```suggestion
                try:
                    response.seek(0)
                except (AttributeError, io.UnsupportedOperation):
                    response = BytesIO(response.read())
                return imread(response, format=ext)
```
is what pillow is doing (https://github.com/python-pillow/Pillow/pull/1151/files).